lib/date-utils.ts

```diff
@@ -0,0 +1,13 @@
+import { isBefore } from "date-fns";
+
+export type TimeInterval = { intervalStartDate: Date; intervalEndDate: Date };
+
+export const doTimeIntervalsOverlap = (
+  intervalA: TimeInterval,
+  intervalB: TimeInterval
+): boolean => {
+  return (
+    isBefore(intervalA.intervalStartDate, intervalB.intervalEndDate) &&
+    isBefore(intervalB.intervalStartDate, intervalA.intervalEndDate)
+  );
```

Comment:
Proof this logic works in image below just because I know we've had a lot of conversations about what exactly the right way to do this is. I'll see if I can switch over the sampling date filter to use this piece of code as well at a later date :)

![20240120_151835](https://github.com/PathoTracker/Pathotracker/assets/86806388/e6d976a3-193d-4a71-a5cb-2e67d03dbdb9)